### PR TITLE
Add Python 3.12 support

### PR DESCRIPTION
**Pull request by tisto**

### CODE REVIEW

**Comment by davisagli** [comment]
@tisto The tests failed. I think you need to update actions/setup-python to a newer version

File: plone-6.0.x.cfg [hunk 1]
```diff
@@ -11,4 +11,10 @@ zodb-temporary-storage = off
 plone.rest =
 black = 23.3.0
 pygments = 2.14.0
-grpcio-tools = 1.59.0
\ No newline at end of file
+
+# all this is neccessary to make the p.a.contenttypes (robotframework) test dependency happy :(
```
Comment (davisagli) [inline plone-6.0.x.cfg:15]: Should these pins be updated in buildout.coredev then?

Comment (tisto) [reply plone-6.0.x.cfg:15]: Philip already took care of this in his PR: https://github.com/plone/buildout.coredev/pull/890

I just copied over those version pins...

Comment (davisagli) [reply plone-6.0.x.cfg:15]: Okay, so probably they are only needed temporarily until that PR is merged. 👍

**Review by mauritsvanrees — APPROVED** [review]
pre-commit was complaining. I fixed it.

LGTM, can be merged when green.

**Comment by davisagli** [comment]
@jenkins-plone-org please run jobs

**Pull request merged**